mnefun/_mnefun.py

```diff
@@ -829,29 +873,128 @@ def run_sss_positions(fname_in, fname_out, host='kasga', opts='', port=22):
             fnames_in.append(next_name)
         else:
             break
-    port = str(int(port))
-    t0 = time()
-    remote_ins = ['~/' + op.basename(fname) for fname in fnames_in]
-    remote_out = '~/temp_%s_raw_quat.fif' % t0
-    remote_hp = '~/temp_%s_hp.txt' % t0
+
     print('  Copying file to %s' % host)
-    cmd = ['scp', '-P' + port] + fnames_in + [host + ':~/']
+    cmd = ['scp', '-P' + str(port)] + fnames_in + [host + ':~/']
     run_subprocess(cmd, stdout=None, stderr=None)
+    t0 = time()
+    remote_ins = ['~/' + op.basename(f) for f in fnames_in]
+    if len(remote_ins) > 1:
+        fnames_out = [op.basename(r)[:-4] + '.tmp' for r in remote_ins]
+        assert len(fnames_out) == len(fnames_in)
+    else:
+        fnames_out = [op.basename(fname_out)[:-4] + '.tmp']
+    for fi, file_out in enumerate(fnames_out):
+        remote_out = '~/temp_%s_raw_quat.fif' % t0
+        remote_hp = '~/temp_%s_hp.txt' % t0
+
+        print('  Running maxfilter as %s' % host)
+        cmd = ['ssh', '-p', str(port), host,
+               '/neuro/bin/util/maxfilter -f ' + remote_ins[fi] + ' -o ' + remote_out +
+               ' -headpos -format short -hp ' + remote_hp + ' ' + opts]
+        run_subprocess(cmd)
 
-    print('  Running maxfilter on %s' % host)
-    cmd = ['ssh', '-p', port, host,
-           'maxfilter -f ' + remote_ins[0] + ' -o ' + remote_out +
-           ' -headpos -format short -hp ' + remote_hp + ' ' + opts]
-    run_subprocess(cmd)
-
-    print('  Copying result to %s' % fname_out)
-    cmd = ['scp', '-P' + port, host + ':' + remote_hp, fname_out]
-    run_subprocess(cmd)
+        print('  Copying result to %s' % file_out)
+        cmd = ['scp', '-P' + str(port), host + ':' + remote_hp, op.join(pout, file_out)]
+        run_subprocess(cmd)
 
-    print('  Cleaning up %s' % host)
-    cmd = ['ssh', '-p', port, host, 'rm -f %s %s %s'
+        print('  Cleaning up %s' % host)
+        cmd = ['ssh', '-p', str(port), host, 'rm -f %s %s %s'
            % (' '.join(remote_ins), remote_hp, remote_out)]
-    run_subprocess(cmd)
+        run_subprocess(cmd)
+
+    # concatenate hp pos file for split raw files if any
+    pos_data = []
+    with open(fname_out, 'w') as fo:
+        for f in fnames_out:
+            with open(op.join(pout, f)) as infile:
+                content = infile.readlines()
+                header = content[0]
+                qdata = content[1:]
+                pos_data.append(qdata)
+                os.remove(op.join(pout, f))
```

Comment:
Only indent as many lines as necessary in the `with` context manager (in this case, I think just the `infile.readlines()` one). Especially this `os.remove` line, it should not be inside a context manager that has that file open.